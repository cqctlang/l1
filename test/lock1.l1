//gc(3,3);
x = cons(10101,0);
y = cons(x,x);
printf("1\n");
gcprotect(x);
x = nil;
printf("2\n");
gc(3,3);
printf("3\n");
gcunprotect(car(y));
printf("4\n");
printf("%a %a\n", car(car(y)), cdr(cdr(y)));
gc(2,2);
printf("5\n");
printf("%a %a\n", car(car(y)), cdr(cdr(y)));
