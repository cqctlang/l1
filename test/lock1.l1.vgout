1
2
3
4
10101 0
5
10101 0
