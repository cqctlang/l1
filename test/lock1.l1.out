entry: var params, 1 locals, 0 temps
1
2
3
4
10101 0
5
10101 0
